transport/ipc/ipc_peer_solaris.go

```diff
@@ -49,14 +50,20 @@ import (
 import "C"
 
 func getPeer(c *net.UnixConn, pipe transport.ConnPipe) {
-	if f, err := c.File(); err == nil {
-		mc := &C.mycred_t{}
-		if C.getucred(C.int(f.Fd()), mc) == 0 {
-			pipe.SetOption(mangos.OptionPeerPID, int(mc.pid))
-			pipe.SetOption(mangos.OptionPeerUID, int(mc.uid))
-			pipe.SetOption(mangos.OptionPeerGID, int(mc.gid))
-			pipe.SetOption(mangos.OptionPeerZone, int(mc.zid))
-		}
+	// This change was necessary to support SetDeadline and Close
+	// which aborts pending reads/writes.
+	// The prior code was calling c.File().Fd(). It was not closing
+	// the file. Additionally the docs say Fd() causes deadlines to not work.
```

Comment:
When I originally wrote this, the Control method didn't exist.  I need to review the docs to understand what is happening here.

Reply:
Here was an attempt to call Close on the connection without this change. Close blocked forever instead of aborting the read.

1 @ 0x43ce36 0x44d0be 0x44d095 0x4654c5 0x4b27cd 0x50ea18 0x519ae5 0x528177 0x4696c1
#       0x4654c4        internal/poll.runtime_Semacquire+0x24                           c:/bsros/toolchain/3.4.1/lib/go/src/runtime/sema.go:67
#       0x4b27cc        internal/poll.(*FD).Close+0x6c                                  c:/bsros/toolchain/3.4.1/lib/go/src/internal/poll/fd_unix.go:116
#       0x50ea17        net.(*netFD).Close+0x37                                         c:/bsros/toolchain/3.4.1/lib/go/src/net/fd_posix.go:37
#       0x519ae4        net.(*conn).Close+0x44                                          c:/bsros/toolchain/3.4.1/lib/go/src/net/net.go:207